cluster-autoscaler/cloudprovider/gce/autoscaling_gce_client_test.go

```diff
@@ -568,3 +568,333 @@ func TestUserAgent(t *testing.T) {
 	assert.NoError(t, err)
 	mock.AssertExpectationsForObjects(t, server)
 }
+
+func TestFetchAllInstances(t *testing.T) {
+	igm1 := "projects/893226960234/zones/zones/instanceGroupManagers/test-igm1-grp"
+	igm2 := "projects/893226960234/zones/zones/instanceGroupManagers/test-igm2-grp"
+	malformedIgm := "projects/893226960234/zones/zones/miss-formed/test-igm1-grp"
+	tests := []struct {
+		name            string
+		liResponse      gce_api.InstanceList
+		liPageResponses map[string]gce_api.InstanceList
+		want            []GceInstance
+	}{
+		{
+			name: "empty response",
+			liResponse: gce_api.InstanceList{
+				Items: []*gce_api.Instance{},
+			},
+			want: nil,
+		},
+		{
+			name: "response with malformed created-by field",
```

Comment:
nit: how about adding a second instance with correct igm to the list to make sure that the malformed igm does not affect parsing another instance.